React Hooksの課題

ham-cap opened the pull request:
React Hooksの課題を提出いたします。
ご確認のほどよろしくお願いいたします🙏

ham-cap committed:
MemosSectionを関数コンポーネント化

ham-cap committed:
useEffectを修正

・既存メモ取得の際に無限ループが発生していたため修正した。

ham-cap committed:
Formを関数コンポーネント化

ham-cap committed:
MemoListを関数コンポーネント化

ham-cap committed:
Memoを関数コンポーネント化

ham-cap committed:
Appを関数コンポーネント化

ham-cap committed:
既存メモ取得の処理をカスタムフックとして切り出し

m0nch1 commented on src/useMemos.jsx:
```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react'
```
このファイルは jsx ではなさそうですね！ js で良さそうです！

m0nch1 commented on src/useMemos.jsx:
```diff
@@ -0,0 +1,23 @@
+  const [memos, setMemos] = useState([])
+
+  const getCurrentMemos = () => {
+    const memosJson = localStorage.getItem('memos')
```
[imo]
この時点ではまだ json にはなっていないと思いますので他の命名が良い気がします！
localStroage から取得したものであることがわかるとより良いかもしれませんね ☺️

m0nch1 commented on src/useMemos.jsx:
```diff
@@ -0,0 +1,23 @@
+    if (memosJson === null) {
+      return []
+    } else {
+      const memosJson = localStorage.getItem('memos')
```
7 行目で取得しているので使いまわせそう？
意図があったら教えてください！

ham-cap replied:
単なる重複です💦
失礼しました🙇‍♂️

m0nch1 commented on src/MemosSection.jsx:
```diff
@@ -1,65 +1,41 @@
-    this.setState({ memos: currentMemos })
-  }
+function MemosSection () {
+  const [isVisible, setIsVisible] = useState(false)
```
ここの state 名は少し可読性あげられそうかなと思いました！
ぱっと見だとなんの visible の状態を保持しているのかわからないので isVisibleForm のようにしても良さそうかと。
今回の場合、〇〇状態の場合にフォームを表示する、という動作のようですので、isEdit とかでもいい気がします！

m0nch1 reviewed (changes requested):
今回の変更箇所メインでレビューしました！
いくつかコメントしてますので確認お願いします！

ham-cap committed:
useMemos.jsxの拡張子を.jsに変更

・JSXを使用していないため。

ham-cap committed:
useMemosを修正

・LocalStorageから取得したデータを格納する変数名をmemosJsonとしていたが、データの取得時点ではJSON形式ではないため修正した。
・LocalStorageからデータを取得するための記述が重複していたため、不要なほうを削除した。

ham-cap committed:
MemosSection内で使用するstateの名称を変更

・isVisibleという名称だと何を可視化するか分かりにくいため、フォームの表示/非表示切り替えに使用することを連想できるよう変更した。

ham-cap commented:
修正してみましたのでご確認をお願いいたします🙇‍♂️

m0nch1 commented on src/useMemos.js:
```diff
@@ -4,12 +4,11 @@ export const useMemos = () => {
   const getCurrentMemos = () => {
-    const memosJson = localStorage.getItem('memos')
-    if (memosJson === null) {
+    const memosFromLocalStorage = localStorage.getItem('memos')
```
👍

m0nch1 reviewed (approved):
修正ありがとうございます！
LGTM 👍
